sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/InputsAdapter.java

```diff
@@ -152,16 +152,23 @@ private Optional<SclReportItem> validateExtRefSource(TExtRef extRef, IEDAdapter
             return warningReportItem(extRef, String.format("The signal ExtRef ExtRefldinst does not match any " +
                 "LDevice with same inst attribute in source IED %s", sourceIed.getXPath()));
         }
-        Optional<String> optionalLDeviceStatus = sourceLDevice.getLDeviceStatus();
-        if (optionalLDeviceStatus.isEmpty()) {
+        Optional<String> optionalSourceLDeviceStatus = sourceLDevice.getLDeviceStatus();
+        if (optionalSourceLDeviceStatus.isEmpty()) {
             return fatalReportItem(extRef, String.format("The signal ExtRef source LDevice %s status is undefined",
                 sourceLDevice.getXPath()));
 
-        } else if (OFF.equals(optionalLDeviceStatus.get())) {
-            return warningReportItem(extRef, String.format("The signal ExtRef source LDevice %s status is off",
-                sourceLDevice.getXPath()));
         }
-        return Optional.empty();
+        String sourceLDeviceStatus = optionalSourceLDeviceStatus.get();
```

Comment:
you could have worked with optionals instead of getting the value and wrapping this value into another optional if you wanted : 
```java
if (optionalSourceLDeviceStatus.isEmpty()) {
            return fatalReportItem(extRef, String.format("The signal ExtRef source LDevice %s status is undefined", sourceLDevice.getXPath()));
        }
return optionalSourceLDeviceStatus
                .map(sourceLDeviceStatus -> {
                    switch (sourceLDeviceStatus) {
                        case OFF:
                            return SclReportItem.warning(extRefXPath(extRef.getDesc()), String.format("The signal ExtRef source LDevice %s status is off", sourceLDevice.getXPath()));
                        case ON:
                            return null;
                        default:
                            return SclReportItem.fatal(extRefXPath(extRef.getDesc()), String.format("The signal ExtRef source LDevice %s status is neither \"on\" nor \"off\"", sourceLDevice.getXPath()));
                    }
                });
```

Reply:
OK